code/game/machinery/Sleeper.dm

```diff
@@ -74,6 +75,15 @@
 					occupant.vessel.trans_to_obj(beaker, pumped + 1)
 		else
 			toggle_filter()
+			
+	if(pump > 0)
```

Comment:
```suggestion
	if(pump)
```
If pump is not equal to 0, then it's always TRUE so you don't need to check for it.